packages/compass-components/src/components/combobox-with-custom-option.tsx

```diff
@@ -44,6 +44,16 @@ export const ComboboxWithCustomOption = <
       {...props}
       multiselect={multiselect}
       onFilter={setSearch}
+      onBlur={() => {
+        // Select an option on blur event to fix missing values on modal submit COMPASS-6511
+        if (!onChange) return;
+        if (search) {
+          if (!userOptions.find((x) => x.value === search)) {
+            setCustomOptions([{ value: search } as K]);
+          }
+          (onChange as onChangeType<false>)(search);
```

Comment:
What happens if `multiselect` is `true`?

Reply:
Added a comment that we skip the onBlur event for multiselect, because in this case, a user should explicitly check the value in the list.